docs/cow-protocol/reference/core/auctions/schema.md

```diff
@@ -15,441 +18,188 @@ To avoid precision loss, some numerical literals are encoded as strings, referre
+1. Token A is a trusted token, i.e., the corresponding `trusted` flag is set to `true`. This means that the protocol is happy to store this token in the settlement contract.
+2. There is enough balance of sell token B, i.e., at least as much amount as the sell amount of the AMM interaction. This is revealed by the `availableBalance` entry in the token description.
 
-An example is given below.
+If both conditions are satisfied, a solver can set the `internalize` flag to `true` in order to internalize the interaction:
 
 ```json
-"interaction_data": [
-    {
-        "target": "0xdef1c0ded9bec7f1a1670819833240f027b25eff",
-        "value": "0x0",
-        "call_data": "0x..."
-        "inputs": [
-            {
-                "amount": "9907695503532850274304",
-                "token": "0x853d955acef822db058eb8505911ed77f175b99e"
-            }
-        ],
-        "outputs": [
-            {
-                "amount": "6188703276265770703",
-                "token": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
-            }
-        ],
-        "exec_plan": {
-            "sequence": 0,
-            "position": 1
-        },
-        "cost": {
-            "amount": "1000000000000000",
-            "token": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
-        }
-    }
-]
+"internalize": true
 ```
 
+In such a case, the default driver will remove the interaction, and so the solution will end up using less gas, get better ranking, and also be risk-free (at least the part involving the internalized AMM interaction).
+
 ### `score`
 
-The score is the "bid" a solver makes for the batch in the [solver auction](rewards), as it will get ranked according to it. The protocol picks the solution with the highest score, given that it is strictly positive. The score is a required stringified integer denominated in wei. 
+The score is a key that describes the "bid" a solver makes for the batch in the [solver auction](rewards), as it will get ranked according to it. The protocol picks the solution with the highest score, given that it is strictly positive. The score maps to a dictionary with two entries. The first entry is the following:
+
+- `kind`: this is a string of the set {"solver", "riskAdjusted"}, that determines whether a solver will explicitly provide a score or will delegate the score computation to the default driver by only specifying a probability of success for the proposed solution.
+
+If we have `"kind": "solver"`, then there is a second entry, labeled `score`, that corresponds to a stringified float that specifies the score attached to the solution. On the other hand, if we have `"kind": "riskAdjusted"`, then there is a second entry, labeled `successProbability`, that is a a stringified float that specifies the success probability of the proposed solution.
```

Comment:
```suggestion
If we have `"kind": "riskAdjusted"`, then there is a second entry, labeled `successProbability`, that is a a stringified float that specifies the success probability of the proposed solution. On the other hand, if we have `"kind": "solver"`, then there is a second entry, labeled `score`, that corresponds to a stringified float that specifies the score attached to the solution.

:::note
All solvers are encouraged to submit a score via `"kind": "riskAdjusted"` and `successProbability`. Submitting scores directly via `"kind": "solver"` is discouraged and this option might get deprecated.
:::
```
Not sure how to formulate this. I think we should explicitly discourage the submission of scores.

Reply:
Added some clarifications. Let me know if you think it's fine now.